lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb

```diff
@@ -112,8 +112,12 @@ def generate_settings_to_import_pod_targets
               build_settings['HEADER_SEARCH_PATHS'] = '$(inherited) ' + XCConfigHelper.quote(library_header_search_paths)
               build_settings['OTHER_CFLAGS'] += ' ' + XCConfigHelper.quote(library_header_search_paths, '-isystem')
             end
-            if pod_targets.any? { |t| t.should_build? && t.scoped? }
-              build_settings['FRAMEWORK_SEARCH_PATHS'] = '"$PODS_FRAMEWORK_BUILD_PATH"'
+            scoped_pod_targets = build_pod_targets.select(&:scoped?)
+            unless scoped_pod_targets.empty?
+              framework_search_paths = scoped_pod_targets.map do |target|
+                target.relative_configuration_build_dir
+              end.uniq
+              build_settings['FRAMEWORK_SEARCH_PATHS'] = XCConfigHelper.quote(framework_search_paths)
```

Comment:
This will cause that the aggregate target inherits `FRAMEWORK_SEARCH_PATHS` from the user project. I see how that would make sense for the user project, but it could cause unexpected issues with the aggregate target through misconfiguration. I don't think that this should belong into this PR.